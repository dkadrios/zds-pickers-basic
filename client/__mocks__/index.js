export inputControls from './inputControls'
export mappings from './mappings'
export midi from './midi'
export shiftGroups from './shiftGroups'
export shifter from './shifter'
export store from './store'
export user from './user'
export version from './version'
