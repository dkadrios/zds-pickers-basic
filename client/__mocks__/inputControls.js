export default [
  {
    controlId: 0,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 11,
    value: 110,
    active: true,
    lit: true,
  },
  {
    controlId: 1,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 1,
    status: 11,
    value: 111,
    active: false,
    lit: false,
  },
  {
    controlId: 2,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: false,
  },
  {
    controlId: 3,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: true,
  },
  {
    controlId: 4,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: true,
  },
  {
    controlId: 5,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: false,
  },
]
