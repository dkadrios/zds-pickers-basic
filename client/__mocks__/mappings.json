{
  "dialogVisible": false,
  "userDialogVisible": false,
  "userImportDialogVisible": false,
  "channels": [
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    "BFD3",
    null,
    null,
    null,
    null,
    null,
    null
  ],
  "stockMappings": [
    "2box",
    "Addictive Drums",
    "Alesis DM Pro",
    "Alesis DM10",
    "Alesis DM5",
    "BFD2",
    "BFD3",
    "Battery 3",
    "E-MU Protean Acoustic",
    "E-MU Protean Bread n Butter",
    "E-MU Protean Studio",
    "E-MU Protean Synth",
    "EZ Drummer The Classic EZX",
    "EZDrummer",
    "EZDrummer Jazz Kit",
    "EZDrummer Latin Percussion",
    "General MIDI",
    "Infinite Instruments",
    "John Emrich - Articulations 1",
    "Kontakt 2",
    "Ocean Way Drums",
    "Pearl Redbox",
    "Roland HandSonic 15",
    "Roland TD-10",
    "Roland TD-10 Perc 1",
    "Roland TD-10 Perc 2",
    "Roland TD-10 Perc 3",
    "Roland TD-10 Perc 4",
    "Roland TD-12",
    "Roland TD-12 Perc 1 (Latin)",
    "Roland TD-12 Perc 2 (Indian)",
    "Roland TD-12 Perc 3 (African)",
    "Roland TD-12 Perc 4 (Salsa)",
    "Roland TD-12 Perc 5 (SFX)",
    "Roland TD-12 Perc 6-8 (User)",
    "Roland TD-20",
    "Roland TD-20 Perc",
    "Roland TD-3",
    "Roland TD-30",
    "Roland TD-30 Perc 1 (Latin)",
    "Roland TD-30 Perc 2 (Indian)",
    "Roland TD-30 Perc 3 (African)",
    "Roland TD-30 Perc 4 (Salsa)",
    "Roland TD-30 Perc 5 (SFX)",
    "Roland TD-30 Perc 6-8 (User)",
    "Roland TD-5",
    "Roland TD-6",
    "Roland TD-6 Perc 1",
    "Roland TD-6 Perc 10 (Special)",
    "Roland TD-6 Perc 2",
    "Roland TD-6 Perc 3 (Room)",
    "Roland TD-6 Perc 4 (Power)",
    "Roland TD-6 Perc 5 (Elec)",
    "Roland TD-6 Perc 6 (808)",
    "Roland TD-6 Perc 7 (Jazz)",
    "Roland TD-6 Perc 8 (Brush)",
    "Roland TD-6 Perc 9 (Perc)",
    "Roland TD-7",
    "Roland TD-8",
    "Roland TD-8 Perc 1",
    "Roland TD-8 Perc 10 (Special)",
    "Roland TD-8 Perc 2",
    "Roland TD-8 Perc 3 (Room)",
    "Roland TD-8 Perc 4 (Power)",
    "Roland TD-8 Perc 5 (Elec)",
    "Roland TD-8 Perc 6 (808)",
    "Roland TD-8 Perc 7 (Jazz)",
    "Roland TD-8 Perc 8 (Brush)",
    "Roland TD-8 Perc 9 (Perc)",
    "Roland TD-9",
    "Roland TD-9 Perc",
    "Simmons SD1000",
    "Simmons SD5X",
    "Simmons SD7",
    "Simmons SD7PK",
    "Simmons SD9K",
    "Sonic Reality iMap Extended",
    "Sonic Reality iMap Standard",
    "Studio ProFiles",
    "Superior Drummer",
    "Yamaha DTX 700",
    "Yamaha DTXplorer",
    "Zendrum Stompblock",
    "Zendrum TRS",
    "Zendrum TRS 2.0"
  ],
  "userMappings": ["BFD3 Custom ZenEdit", "reaper"]
}
