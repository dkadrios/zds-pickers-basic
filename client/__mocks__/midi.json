{
  "devices": [
    {
      "id": "-1325431060",
      "manufacturer": "Zendrum Studio",
      "name": "ZDS Shifter",
      "type": "output",
      "version": "",
      "state": "connected",
      "connection": "open"
    },
    {
      "id": "-312145282",
      "manufacturer": "Zendrum Studio",
      "name": "ZDS Shifter",
      "type": "input",
      "version": "",
      "state": "connected",
      "connection": "open"
    }
  ],
  "listeningDevices": ["-312145282"]
}
