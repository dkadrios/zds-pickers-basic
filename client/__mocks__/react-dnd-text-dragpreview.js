/* eslint-disable import/prefer-default-export, jsx-a11y/alt-text */
import React from 'react'

export const createDragPreview = () => <img src={null} />
