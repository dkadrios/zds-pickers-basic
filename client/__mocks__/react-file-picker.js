/* eslint-disable import/prefer-default-export */
export const FilePicker = () => <div>content</div>
