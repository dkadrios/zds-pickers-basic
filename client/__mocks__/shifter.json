{
  "accessGranted": true,
  "errorMessage": "",
  "errorVisible": false,
  "exportBuffer": [],
  "exportDialogVisible": false,
  "exportFilename": "zds-shifter-backup.txt",
  "found": true,
  "hardwareTestVisible": true,
  "importDialogVisible": false,
  "importInProcess": false,
  "invalidSettingsFile": null,
  "midiActivityLEDMode": 0,
  "midiInActivity": false,
  "midiOutActivity": false,
  "ready": true,
  "resetInProcess": false,
  "responding": true,
  "searchedForShifter": true,
  "serialMidiOutEnabled": true,
  "showResetDialog": false,
  "testInterfaceFound": false,
  "usbMidiOutEnabled": true
}
