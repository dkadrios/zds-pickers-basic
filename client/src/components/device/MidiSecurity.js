import React, { Component } from 'react'
import { connect } from 'react-redux'
import DialogTitle from '@material-ui/core/DialogTitle'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import { shifterShape } from '../../core/shapes'
import Dialog from '../Dialog'

export class MidiSecurity extends Component {
  constructor(props) {
    super(props)
    this.state = { active: false }
  }

  // eslint-disable-next-line camelcase
  UNSAFE_componentWillMount() {
    const { shifter } = this.props
    // Give web midi a chance to start up before assuming no access given.
    setTimeout(
      /* istanbul ignore next */ () => {
        this.setState({ active: !shifter.accessGranted })
      },
      2000,
    )
  }

  render() {
    const { shifter } = this.props
    const { active } = this.state

    return (
      <Dialog open={Boolean(active) && !shifter.accessGranted}>
        <DialogTitle>Your Permission Is Required</DialogTitle>
        <DialogContent>
          <DialogContentText>
            This application requires special permissions before it can use SysEx and connect to your ZDS Shifter.
          </DialogContentText>

          <DialogContentText>
            <br />
          </DialogContentText>

          <DialogContentText>Please select &apos;Allow&apos; when prompted by your browser.</DialogContentText>
        </DialogContent>
      </Dialog>
    )
  }
}

MidiSecurity.propTypes = {
  shifter: shifterShape.isRequired,
}

export const mapStateToProps = ({ shifter }) => ({ shifter })

export default connect(mapStateToProps)(MidiSecurity)
