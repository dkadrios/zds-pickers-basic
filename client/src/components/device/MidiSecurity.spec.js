import React from 'react'
import set from 'lodash/fp/set'
import { shifter } from '../../../__mocks__'
import { MidiSecurity, mapStateToProps } from './MidiSecurity'

const props = {
  shifter,
}

describe('MidiSecurity tests', () => {
  beforeEach(jest.resetAllMocks)

  it('renders correctly', () => {
    shallowExpect(MidiSecurity)(props).toMatchSnapshot()
  })

  it('renders correctly when shifter is available but we can`t access it', () => {
    const wrapper = shallow(<MidiSecurity {...set('shifter.accessGranted', false)(props)} />)
    wrapper.setState({ active: true })
    expect(wrapper).toMatchSnapshot()
  })

  it('should utilize mapStateToProps', () => {
    const state = { shifter }
    expect(mapStateToProps(state)).toEqual(state)
  })
})
