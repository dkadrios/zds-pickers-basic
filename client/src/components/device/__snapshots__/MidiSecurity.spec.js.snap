// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MidiSecurity tests renders correctly 1`] = `
<Dialog
  open={false}
>
  <WithStyles(DialogTitle)>
    Your Permission Is Required
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <WithStyles(DialogContentText)>
      This application requires special permissions before it can use SysEx and connect to your ZDS Shifter.
    </WithStyles(DialogContentText)>
    <WithStyles(DialogContentText)>
      <br />
    </WithStyles(DialogContentText)>
    <WithStyles(DialogContentText)>
      Please select 'Allow' when prompted by your browser.
    </WithStyles(DialogContentText)>
  </WithStyles(DialogContent)>
</Dialog>
`;

exports[`MidiSecurity tests renders correctly when shifter is available but we can\`t access it 1`] = `
<Dialog
  open={true}
>
  <WithStyles(DialogTitle)>
    Your Permission Is Required
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <WithStyles(DialogContentText)>
      This application requires special permissions before it can use SysEx and connect to your ZDS Shifter.
    </WithStyles(DialogContentText)>
    <WithStyles(DialogContentText)>
      <br />
    </WithStyles(DialogContentText)>
    <WithStyles(DialogContentText)>
      Please select 'Allow' when prompted by your browser.
    </WithStyles(DialogContentText)>
  </WithStyles(DialogContent)>
</Dialog>
`;
